core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -429,20 +429,40 @@ private[ui] class JobDataSource(
     val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
     val submissionTime = jobData.submissionTime
     val formattedSubmissionTime = submissionTime.map(UIUtils.formatDate).getOrElse("Unknown")
-    val jobDescription = UIUtils.makeDescription(jobData.description.getOrElse(""),
-      basePath, plainText = false)
+
+    val lastStage = {
+      val stageAttempts = jobData.stageIds.flatMap(store.stageData(_))
+      if (!stageAttempts.isEmpty) {
+        val lastId = stageAttempts.map(_.stageId).max
+        stageAttempts.find(_.stageId == lastId)
```

Comment:
I think these two lines can just be `Some(stageAttempts.maxBy(_.stageId))`

Reply:
Indeed, nice catch, thx